File: android/nativepreview/src/main/java/com/travelplaner/nativepreview/domain/MapCameraSnapshot.kt
package com.travelplaner.nativepreview.domain

/** Viewport preference only, never written into travel JSON or location history. */
data class MapCameraSnapshot private constructor(val values:List<Double>) {
    companion object {
        fun from(values:List<Double>):MapCameraSnapshot? = values.takeIf {it.size==5 && it.all(Double::isFinite) && it[0] in -90.0..90.0 && it[1] in -180.0..180.0 && it[2] in 0.0..22.0 && it[3] in 0.0..360.0 && it[4] in 0.0..90.0}?.let {MapCameraSnapshot(it.toList())}
    }
}

File: ios/NativeCore/Sources/NativeCore/MapCameraSnapshot.swift
import Foundation

/// Viewport preference only. Never part of a trip, location history, or Google response cache.
public struct MapCameraSnapshot:Equatable {
    public let values:[Double]
    public init?(values:[Double]) {
        guard values.count==5,values.allSatisfy(\.isFinite),(-90...90).contains(values[0]),(-180...180).contains(values[1]),
              (0...22).contains(values[2]),(0...360).contains(values[3]),(0...90).contains(values[4]) else {return nil}
        self.values=values
    }
}
